src/hw/actions/app.js

```diff
@@ -46,7 +47,7 @@ type State = {|
   isLoading: boolean,
   requestQuitApp: boolean,
   requestOpenApp: ?string,
-  requiresAppInstallation: ?{ appName: string },
+  requiresAppInstallation: ?{ appName: string } | ?{ appNames: string[] },
```

Comment:
in one side you try not to be too breaking (splitting into two actions) but on the state itself, i feel it's actually breaking (if UI side expects .appName)
so i have feeling we should just move to `Array<*>` paradigm everywhere to avoid all the ifs.

Reply:
Makes sense